airflow_provider_fivetran/hooks/fivetran.py

```diff
@@ -0,0 +1,242 @@
+                host = self._parse_host(self.fivetran_conn.extra_dejson['host'])
+            else:
+                host = self.fivetran_conn.host
+        else:
+            self.log.info('Using basic auth. ')
+            auth = (self.fivetran_conn.login, self.fivetran_conn.password)
+            host = self.fivetran_conn.host
+
+        url = host + connector_id + force
+
+        if method == 'GET':
+            request_func = requests.get
+        elif method == 'POST':
+            request_func = requests.post
+        elif method == 'PATCH':
+            request_func = requests.patch
+        else:
+            raise AirflowException('Unexpected HTTP Method: ' + method)
+
+        attempt_num = 1
+        while True:
+            try:
+                response = request_func(
+                    url,
+                    data=json if method == 'PATCH' else None,
+                    params=json if method == 'GET' else None,
+                    auth=auth,
+                    headers={"Content-Type": "application/json;version=2"} if method == 'PATCH' else None,
+                )
+                response.raise_for_status()
+                return response.json()
+            except requests_exceptions.RequestException as e:
+                if not _retryable_error(e):
+                    # In this case, the user probably made a mistake.
+                    # Don't retry.
+                    raise AirflowException(
+                        f'Response: {e.response.content}, Status Code: {e.response.status_code}'
+                    )
+
+                self._log_request_error(attempt_num, e)
+
+            if attempt_num == self.retry_limit:
+                raise AirflowException(
+                    ('API requests to Fivetran failed {} times. ' + 'Giving up.').format(self.retry_limit)
+                )
+
+            attempt_num += 1
+            sleep(self.retry_delay)
+
+    def _log_request_error(self, attempt_num: int, error: str) -> None:
+        self.log.error('Attempt %s API Request to Fivetran failed with reason: %s', attempt_num, error)
+
+    def check_connector(self, connector_id):
+        # Make sure connector configuration has been completed successfully and is not broken.
+        
+        resp = self._do_api_call(('GET',connector_id))
+        connector_details = resp["data"]
+        URL_LOGS = "https://fivetran.com/dashboard/connectors/{}/{}/logs".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        URL_SETUP = "https://fivetran.com/dashboard/connectors/{}/{}/setup".format(
+            connector_details["service"], connector_details["schema"]
+        )
+        setup_state = connector_details["status"]["setup_state"]
+        if setup_state != "connected":
+            EXC_SETUP: str = (
+                'Fivetran connector "{}" not correctly configured, status: {}; '
+                + "please complete setup at {}"
+            )
+            raise ValueError(
```

Comment:
For Astronomer: does it make sense to raise exceptions like this to kill an Operator and fail? If so, should we be raising specific Airflow exceptions? (I am guessing the answer is Yes :)